.github/workflows/update_index.yml

```diff
@@ -0,0 +1,67 @@
+name: Update rules index on registry changed
+
+on:
+  push:
+    branches: [main]
+    paths: registry.yaml
+
+jobs:
+  update-rules-registry:
+    runs-on: ubuntu-latest
+    env:
+      GH_INDEX_REPO: falcoctl
+      GH_INDEX_REPO_BRANCH: gh-pages
+      REGISTRY_USER: ${{ github.repository_owner }}
+      REGISTRY: ghcr.io
+      OCI_REPO_PREFIX: ${{ env.REGISTRY }}/${{ env.REGISTRY_USER }}/${{ github.event.repository.name }}
+
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: write
+      packages: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout Rules
+        uses: actions/checkout@v3
+
+      - name: Git set-up
+        run: |
+          apt-get update && apt-get install -y --no-install-recommends git && apt-get clean
+
+      - name: Setup Golang
+        uses: actions/setup-go@v3
```

Comment:
there is a v4 version available for `actions/setup-go` if we can use the most updated one.